### DISCUSSION TITLE
Nested FieldArray  disappear when changes step in form

### DISCUSSION THREAD

**Discussion by Kalgros20** [react-hook-form/react-hook-form#5589] [Q&A]
Hi guys, first of all, thanks for the awesome lib. 

I don't know if my problem it's more about my react knowledge, or a wrong usage of the lib... But let's go the point.

I have a 3 step form, and i'm using material-ui stepper with RHF.

I'm using the FormContextProvider  + useFormContext because i have many levels of inputs... My problem happens, when i need to get the async default values from an API,  i have  async default values in different steps. So, i just execute the useEffect with the request + reset when thactivestep changes.  In the second step, i havee  a nested field array with values that comes from an api, and the array disappear in render. 

Steps to reproduce:
1 - Click Next button.
    See the nested array blinks and dissapear.

Here's the [codesandbox](https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-forked-dr507?file=/src/index.js)

**Comment by bluebill1049** [comment]
any chance you can upgrade to the latest v7? v7 has a better architecture for field array and it performs a lot better as well.

**Reply by Kalgros20** [reply]
Tested with v7  [here](https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-forked-dr507?file=/src/index.js) and worked fine. I think that i can do the upgrade. When i finish the upgrade i come back to tell you the result.

**Reply by bluebill1049** [reply]
👍
